nbformat/tests/test_sign.py

```diff
@@ -28,7 +28,16 @@ def setUp(self):
     
     def tearDown(self):
         shutil.rmtree(self.data_dir)
-    
+   
+    def test_invalid_db_file(self):
+        invalid_notary = sign.NotebookNotary(
+            db_file='invalid_db_file.db',
+            secret=b'secret',
+            data_dir='.',
```

Comment:
I think you have to do something with the NotebookNotary that will make it open the database. The error in ipython/ipython#9003 is from calling the `check_signature` method. It doesn't open the database when you instantiate it, because of the way traitlets work.